### PR TITLE
feat(entity-schema): merged all-packages view for get-entity-schema-properties when --package omitted

### PR DESCRIPTION
**Pull request by Alexandr-Kravchuk**
## Problem

`get-entity-schema-properties` required `--package` and returned only **that package layer's** column slice. Reading `Account` in `Base` reported `own=0, inherited=33` and never surfaced custom columns added in *other* packages — e.g. `UsrColumn40` (Contract lookup) in `UsrCrtLeadOppMgmtApp_*`. An AI agent therefore concluded the field did not exist and proposed creating a column that already existed.

**Root cause:** `RemoteEntitySchemaColumnManager.GetSchemaProperties` resolves a single package and calls `GetSchemaDesignItem(..., UseFullHierarchy=false)`, so customizations from other packages are never returned.

## Change — Variant A (make `--package` optional)

- **`--package` omitted** → returns the **merged / effective** schema via the existing `IRuntimeEntitySchemaReader.GetByName(schemaName)`, which queries the runtime endpoint **by name (no package)** and returns the complete column set across all packages. `package-name` reports `(merged: all packages)`.
- **`--package` supplied** → unchanged single-package-layer slice (**backward compatible**, keeps parent schema / indexes / SSP and other schema-level metadata).

Why the runtime reader: it already exists, is DI-registered, and is used by DataForge; its runtime `DataValueType` ids map 1:1 through `GetFriendlyTypeName`, so friendly type names stay consistent across both modes. `IsInherited` maps to `own`/`inherited`; the primary column is resolved via `PrimaryColumnUId`. No new endpoint and no guessing the "top" package.

## MCP reviewed

- Tool + args descriptions updated; `GetEntitySchemaPropertiesArgs.package-name` is now **optional** (kebab-case JSON preserved).
- `get-entity-schema-properties` prompt: `package-name` optional + explicit **"an empty single-package read does NOT prove a column is absent"** guidance (re-read without package, or use `find-entity-schema`).
- Guidance resources updated to steer toward the merged read: `ExistingAppMaintenance`, `Esq`, `AppModeling`.
- **E2E added** in `clio.mcp.e2e/EntitySchemaToolE2ETests.cs` (merged-mode test + null-package call helper). Note: `clio.mcp.e2e` is **not in CI** and needs a real sandbox, so it was not executed here.
- Tool safety flags unchanged (`ReadOnly=true, Destructive=false, Idempotent=true`).

## Docs

`clio/help/en/get-entity-schema-properties.txt`, `clio/docs/commands/get-entity-schema-properties.md`, `clio/Commands.md` updated for the two modes and the "empty ≠ absent" caveat.

## Tests / validation

`Validated: dotnet test --filter "Category=Unit&(Module=Command|Module=McpServer)"`
- New unit tests: manager merged-mode + single-layer routing, command merged-acceptance (package omitted is valid), tool null-package mapping, prompt assertions.
- Entity-schema (207) + guidance/contract (175) suites green. The only 3 failures (`ListCreatioBuilds*`, `CreateUiProject_Should_Switch_Working_Directory…`) **pre-exist on clean `master`** (verified via stash) and are unrelated.
- `clio` and `clio.mcp.e2e` build with **0 errors** → no new `CLIO*` analyzer diagnostics.

## Reviewer notes

- ⚠️ `EntitySchemaPrompt.GetEntitySchemaProperties` parameter order changed to `(schemaName, environmentName, packageName = null)`; its unit test was updated.
- Acceptance not yet verified against a live environment containing `Account.UsrColumn40` — recommend a manual `clio get-entity-schema-properties -e <env> --schema-name Account` before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs [hunk 1]
```diff
@@ -206,6 +218,63 @@ public EntitySchemaPropertiesInfo GetSchemaProperties(GetEntitySchemaPropertiesO
...
+	private static EntitySchemaPropertyColumnInfo MapRuntimePropertyColumn(RuntimeEntitySchemaColumnResult column) {
+		return new EntitySchemaPropertyColumnInfo(
+			column.Name,
+			column.UId,
+			column.IsInherited ? "inherited" : "own",
+			column.Caption,
+			column.Description,
+			EntitySchemaDesignerSupport.GetFriendlyTypeName(column.DataValueType),
+			column.IsRequired,
+			false,
```
Comment (Alexandr-Kravchuk) [inline clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: Major — merged mode silently misreports per-column `indexed` and the schema-level flags. Here `indexed` is hardcoded `false` for every column (the runtime reader does not expose it), while the single-package path returns the real `column.Indexed`. Combined with the schema-level fields hardcoded just above (`parent-schema-name`, `extend-parent`, `indexes-count`, `ssp-available`, `track-changes-in-db`, the administration flags, etc.), a consumer cannot distinguish a genuine `false`/`null` from "not populated in this mode." The same JSON field carries real data in single-package mode and a placeholder here, so a column that is genuinely indexed reports `indexed: false` in the merged view and `indexed: true` in the single-package view. Since the tool/prompt explicitly steer agents toward merged mode for column discovery, this is exactly the kind of false signal that leads to wrong conclusions (the bug class this PR fixes). Fix: map `Indexed` through `RuntimeEntitySchemaColumnResult` if the runtime payload exposes it; otherwise document at the record/tool level that schema-level flags and per-column `indexed` are not authoritative in merged mode (consider `bool?` for `indexed` so merged mode can emit `null` = unknown rather than a false negative).

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs [hunk 2]
```diff
@@ -206,6 +218,63 @@ public EntitySchemaPropertiesInfo GetSchemaProperties(GetEntitySchemaPropertiesO
...
+		}
+		RuntimeEntitySchemaResult runtimeSchema = _runtimeEntitySchemaReader.GetByName(options.SchemaName.Trim());
+		List<EntitySchemaPropertyColumnInfo> columns = runtimeSchema.Columns
+			.Select(MapRuntimePropertyColumn)
+			.ToList();
+		int ownColumnCount = columns.Count(column => column.Source == "own");
+		int inheritedColumnCount = columns.Count - ownColumnCount;
+		string? primaryColumnName = runtimeSchema.Columns
+			.FirstOrDefault(column => column.UId == runtimeSchema.PrimaryColumnUId)?.Name;
+		return new EntitySchemaPropertiesInfo(
```
Comment (Alexandr-Kravchuk) [inline clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs:254]: Major (maintainability) — this constructs the 22-parameter `EntitySchemaPropertiesInfo` positionally with a run of 11 consecutive bare `false` literals (lines 251-261) plus several `null`/`0`. The order is correct today, but any future field insertion/reorder in the record will silently keep compiling and shift every boolean into the wrong JSON field — a `false→false` slot shift is undetectable, and no test pins the merged-mode field contract. The single-package builder above is safer because each slot is a named `schema.X` expression. Fix: use named arguments for the literal-heavy fields (`title: null, parentSchemaName: null, extendParent: false, indexesCount: 0, sspAvailable: false, …`) so a future reorder either fails to compile or stays correct — zero runtime cost. Optionally add a `EntitySchemaPropertiesInfo.ForMergedSchema(...)` factory that only takes the fields the runtime read can populate and defaults the rest, centralizing the "not available in merged mode" contract.

**Review by Alexandr-Kravchuk — COMMENTED** [review]
## Code Review — Decision: REQUEST CHANGES

Multi-agent review (code-quality, security, performance, testing, bug-risk). Local review against base `master` (`2acdd1f1` → head `188b6695`). No Playwright files touched — overlay not applicable. Security pass: clean.

### Findings (highest → lowest)

**Major — Merged mode silently misreports `indexed` and schema-level flags** · `RemoteEntitySchemaColumnManager.cs:269-274` + `:239-262`
In merged mode every column's `indexed` is hardcoded `false` (single-package returns real `column.Indexed`), and `parent-schema-name`, `extend-parent`, `indexes-count`, `ssp-available`, `track-changes-in-db`, plus all administration flags are hardcoded `null`/`false`/`0`. These are indistinguishable from genuine values, and the same JSON field carries real data in single-package mode. The tool/prompt steer agents toward merged mode for column discovery, so an agent reading `indexed: false` / `ssp-available: false` can draw the wrong conclusion — the exact failure class this PR fixes. *Fix:* map the fields if the runtime payload exposes them; otherwise document that schema-level flags and per-column `indexed` are not authoritative in merged mode (consider `bool?` for `indexed`).

**Major — Positional construction of the 22-arg record is fragile** · `RemoteEntitySchemaColumnManager.cs:239-262`
11 consecutive bare `false` literals bind positionally; a future field insert/reorder shifts booleans into the wrong field and still compiles, and no test pins the merged field contract. *Fix:* named arguments, or a `ForMergedSchema(...)` factory.

**Major — Negative-path and contract test coverage missing** · `clio.tests/Command/RemoteEntitySchemaColumnManagerTests.cs`, `GetEntitySchemaPropertiesCommandTests.cs`
No test drives the merged path when `GetByName` throws `InvalidOperationException` (the most likely real failure: schema not found → exit code 1), no test pins the hardcoded merged schema-level fields, and whitespace-only `--package` routing (`IsNullOrWhiteSpace`) is unverified. *Fix:* add the throw test, a `[TestCase(null/""/"   ")]` routing test, and assertions on the placeholder fields + `SchemaName.Trim()`.

**Minor — Exception-type divergence on the MCP path** · `RuntimeEntitySchemaReader.cs:69-84`
Single-package failures surface as `EntitySchemaDesignerException`; merged failures surface as raw `InvalidOperationException`. The MCP tool calls `GetSchemaProperties` directly (bypassing the command try/catch), so the exception type differs for the same logical op. *Fix:* translate to `EntitySchemaDesignerException` in `GetMergedSchemaProperties`; the duplicate schema-name guard there is dead code.

**Minor** — own/inherited count re-parses the `"own"`/`"inherited"` string it just serialized from `IsInherited` (`:235`); count directly from `IsInherited`. **Minor** — merged columns are alphabetically ordered while single-package preserves designer order — undocumented divergence. **Minor** — `GetEntitySchemaPropertiesArgs` de-inheritance from `EntitySchemaTargetArgsBase` is correct (package optional) but duplicates two property declarations; add a one-line rationale comment and consider `string? PackageName`.

### Review Summary
Decision: **request changes**. The merged-view feature is well-structured, correctly wired (DI-registered reader, kebab-case JSON, docs + MCP surface updated), and security-clean; bug-detector verified all positional constructor orders are currently correct. The blocking concern is that merged mode emits placeholder `false`/`null` for schema-level metadata and per-column `indexed` indistinguishably from real values — misleading for the AI column-discovery use case this tool targets — compounded by a fragile positional constructor and missing negative-path/contract tests. All findings are addressable; none are deep design flaws.

### Requirement Coverage
- `--package` omitted → merged all-packages view: **covered** (manager/command/MCP unit tests).
- `--package` supplied → unchanged single-package slice: **covered**.
- MCP surface + docs updated for two modes and "empty ≠ absent": **covered**.
- E2E for merged mode: **unclear** — `clio.mcp.e2e` is not in CI and was not executed; live verification against an env containing `Account.UsrColumn40` is still pending (per PR notes).
- Accuracy of non-column schema metadata in merged mode: **missing** (see Major #1).

Confidence: high (full local diff + supporting code reviewed). Coverage limit: merged behavior not validated against a live environment.

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs (hunk 1 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: Addressed in fde85ed4: merged mode now maps per-column `indexed` (`column.IsIndexed`) and the schema-level flags (extend-parent, track-changes-in-db, db-view, virtual, show-in-advanced-mode, administration flags, title/description) from the real runtime payload instead of hardcoded false; the fields the by-name runtime endpoint does not expose (parent-schema-name, indexes-count, ssp-available, use-record-deactivation, use-deny-record-rights, use-live-editing) are documented as non-authoritative in merged mode in the help/docs.

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs (hunk 2 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs:254]: Addressed in fde85ed4: the merged-mode `EntitySchemaPropertiesInfo` construction now uses named arguments for every field (`Title:`, `ExtendParent:`, `IndexesCount: 0`, `SspAvailable: false`, …), so a future record field insertion/reorder either fails to compile or stays correct instead of silently shifting a boolean into the wrong JSON slot.

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs [hunk 3]
```diff
@@ -206,6 +218,83 @@ public EntitySchemaPropertiesInfo GetSchemaProperties(GetEntitySchemaPropertiesO
...
+	}
+
+	/// <summary>
+	/// Reads the runtime schema for the merged view, translating low-level reader failures into the domain
+	/// <see cref="EntitySchemaDesignerException"/> so both read paths surface a uniform exception type.
+	/// </summary>
+	private RuntimeEntitySchemaResult ReadMergedRuntimeSchema(string schemaName) {
+		try {
+			return _runtimeEntitySchemaReader.GetByName(schemaName);
+		} catch (InvalidOperationException exception) {
```
Comment (b-horodyskyi) [inline clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: **Major — incomplete exception translation leaks raw faults on the MCP path.**

This `catch` only translates `InvalidOperationException`. `IRuntimeEntitySchemaReader.GetByName` reaches `IApplicationClient.ExecutePostRequest` and `JsonSerializer.Deserialize`, which can throw `HttpRequestException`, `TaskCanceledException`, or `JsonException` (e.g. network blip, timeout, or an HTML error page instead of JSON). Those bypass this translation.

Why it matters: the `get-entity-schema-properties` MCP tool returns a typed result and calls the resolved command's `GetSchemaProperties(options)` directly — it does **not** go through `BaseTool.InternalExecute`'s catch-all. So on the MCP surface (now the *recommended default* merged mode) a transport/parse failure propagates as an unstructured fault rather than a clean tool error. The CLI path is masked by `Command.Execute`'s broad catch, so only MCP is exposed. The deliberately narrow try/catch here signals an intent to normalize failures that is left incomplete.

Fix: broaden the catch to also translate the realistic transport/parse types into `EntitySchemaDesignerException`, e.g. `catch (Exception ex) when (ex is InvalidOperationException or HttpRequestException or JsonException or TaskCanceledException)`. Enumerate the types — do not use a bare `catch (Exception)` (project rule).

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs [hunk 4]
```diff
@@ -206,6 +218,83 @@ public EntitySchemaPropertiesInfo GetSchemaProperties(GetEntitySchemaPropertiesO
...
+			ParentSchemaName: null,
+			ExtendParent: runtimeSchema.ExtendParent,
+			PrimaryColumnName: primaryColumnName,
+			PrimaryDisplayColumnName: runtimeSchema.PrimaryDisplayColumnName,
+			OwnColumnCount: ownColumnCount,
+			InheritedColumnCount: inheritedColumnCount,
+			IndexesCount: 0,
+			TrackChangesInDb: runtimeSchema.IsTrackChangesInDB,
+			DbView: runtimeSchema.IsDBView,
+			SspAvailable: false,
```
Comment (b-horodyskyi) [inline clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: **Major (design) — six schema-level fields emit silent plausible defaults in the structured JSON, contradicting this PR's own goal.**

In merged mode `SspAvailable`, `UseRecordDeactivation`, `UseDenyRecordRights`, `UseLiveEditing` are hardcoded `false`, `IndexesCount` is `0`, and `ParentSchemaName` is `null` (here and the surrounding block). They serialize into the same `EntitySchemaPropertiesInfo` shape as the authoritative single-package read, as ordinary values. A machine consumer (the MCP agent this feature exists to serve) cannot distinguish "genuinely false/0/null" from "not available in this mode" — and for the same schema a single-package read may report `ssp-available: true` while the merged read reports `false`. That is exactly the contradictory-data-for-the-same-object failure class the PR sets out to eliminate (you correctly fixed it for per-column `indexed` and most schema flags).

The only machine-detectable signal today is string-matching `package-name == "(merged: all packages)"`, which is fragile. The prose caveats (remarks/tool description/help/docs) are reliable for a human but not for a JSON consumer.

Fix options: (a) emit `null` for the six unavailable fields in merged mode so absence is machine-distinguishable (larger record change), or (b) explicitly accept this as documented debt in the PR description and state that programmatic callers must key off the merged package label. Either is fine, but the acceptance should be explicit rather than implicit.

**Review by b-horodyskyi — CHANGES_REQUESTED** [review]
## Creatio Code Review — PR #678 (multi-agent, 5 dimensions + bug detection)

**Decision: REQUEST CHANGES** — one high-confidence Major (error handling) plus a Major design inconsistency. The core feature is correct, well-tested, backward-compatible, and the docs/MCP/E2E coverage is exemplary. The blocking items are small and localized.

### Findings (highest first)

**Major — incomplete exception translation leaks raw faults on the MCP path** · `RemoteEntitySchemaColumnManager.cs:280` (inline)
`ReadMergedRuntimeSchema` only wraps `InvalidOperationException`; `HttpRequestException`/`JsonException`/`TaskCanceledException` from `ExecutePostRequest`/`Deserialize` escape. The MCP tool returns a typed result and calls the command directly, bypassing `BaseTool.InternalExecute`'s catch-all, so on the now-default merged path a transport/parse failure surfaces as an unstructured MCP fault. CLI is masked by its broad catch. Fix: enumerate transport/parse types in the catch (not a bare `catch (Exception)`). Flagged independently by code-quality and bug-detection.

**Major (design) — six merged-mode fields emit silent plausible defaults in structured JSON** · `RemoteEntitySchemaColumnManager.cs:258-269` (inline)
`ssp-available`, `use-record-deactivation`, `use-deny-record-rights`, `use-live-editing` (false), `indexes-count` (0), `parent-schema-name` (null) are indistinguishable from authoritative values to a JSON consumer, and can contradict the single-package read for the same schema — the exact wrong-conclusion class the PR fixes for columns. Either emit `null` for these in merged mode, or explicitly accept it as documented debt in the PR description (the only machine signal today is string-matching the `(merged: all packages)` label).

**Major (test coverage) — two verified-safe branches are untested**
Production logic is null-safe (confirmed), but: (1) primary-column resolution when `PrimaryColumnUId` matches no column (`?.Name` → null) and (2) the empty merged column set (own/inherited counting boundary) have no tests. Add both cheap boundary tests so a future refactor (e.g. `FirstOrDefault`→`First`) can't regress silently. Friendly-type mapping is tested only for `Lookup`; a characterization test for the known raw-id fallthrough (Phone/Web — pre-existing, follow-up already spawned) would anchor the deferred fix.

**Minor**
- Redundant schema-name guard in `GetMergedSchemaProperties` throws `EntitySchemaDesignerException` while command `Validate` throws `ArgumentException` for the identical (already-unreachable) condition; trim is also duplicated. Consider dropping the manager guard or aligning the type.
- Test constructions of the 22-field `EntitySchemaPropertiesInfo` use long bare-positional `false, false, ...` runs (`GetEntitySchemaPropertiesCommandTests.cs`, `EntitySchemaToolTests.cs`); production uses named args — mirror that or use a builder.
- `Print` merged test asserts only the package label + parent placeholder; it does not assert own/inherited column grouping renders. New test naming omits `Should` (consistent with the fixture, but deviates from the documented `MethodName_Should..._When...` rule).

### Dimension results
- **Security: PASS.** Schema name travels in an escaped JSON body via `IApplicationClient` (no injection); merged read uses the standard authenticated DataService route, not a ClioGate/privilege-bypass path (no permission-boundary bypass); specific catch surfaces only a sanitized message, no stack trace. Only residual is the long-standing relay of server error text — not introduced here.
- **Performance: PASS.** Merged mode = exactly one remote round-trip; the 3 in-memory passes over an already-materialized column list are sub-millisecond and dwarfed by the HTTP call. No N+1, no actionable allocation concern.
- **Code quality: good.** Named-arg record construction, appended-with-defaults record evolution, the `<remarks>` justifying the `EntitySchemaTargetArgsBase` divergence, and full docs/MCP/E2E updates are all strong.

### Requirement coverage
- Merged/effective view when `--package` omitted: **covered** (manager routing + mapping + tests).
- Backward-compatible single-package path when `--package` supplied: **covered** (untouched path + routing test).
- MCP tool/prompt/guidance + docs/help updated: **covered** (tool, prompt, 3 resources, help, docs, Commands.md, E2E).
- "empty single-package read ≠ column absent" guidance: **covered** (prompt + tool description + resources + docs + prompt assertions).

### Confidence limits
Reviewed against the PR diff + source at head `535cf47`; the PR branch is not checked out locally. No live Creatio environment was exercised — acceptance against a real schema containing `Account.UsrColumn40` (the PR's motivating case) and the actual server error-message content are unverified. `clio.mcp.e2e` is **not in CI**, so the merged-mode E2E assertions are documentation-grade; CI enforcement of the indexed/title mapping rests solely on the unit tests. Playwright overlay not applicable (no TestKit/page-object/fixture files touched).

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs (hunk 3 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: Fixed in 72ece898: broadened the catch in `ReadMergedRuntimeSchema` to a `when` filter over the realistic transport/parse types — `catch (Exception ex) when (ex is InvalidOperationException or HttpRequestException or JsonException or TaskCanceledException)` — so the MCP merged path (which bypasses BaseTool's catch-all) returns a normalized `EntitySchemaDesignerException` instead of a raw fault. Enumerated, not a bare catch. Added a parameterized test asserting each of the three transport/parse types is translated (inner exception preserved).

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs (hunk 4 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs]: Fixed in 72ece898 via option (a): the six fields the by-name runtime endpoint does not expose are now nullable on `EntitySchemaPropertiesInfo` (`indexes-count`, `ssp-available`, `use-record-deactivation`, `use-deny-record-rights`, `use-live-editing` + already-nullable `parent-schema-name`) and emit `null` in merged mode, so a JSON consumer can distinguish 'unavailable in merged mode' from a genuine false/0. Single-package mode is unchanged (always non-null). CLI prints null as `<unknown>`. Record remarks, tool description, help and docs updated accordingly; the merged-defaults test now asserts null.

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs [hunk 5]
```diff
@@ -206,6 +221,96 @@ public EntitySchemaPropertiesInfo GetSchemaProperties(GetEntitySchemaPropertiesO
...
+	/// </returns>
+	private EntitySchemaPropertiesInfo GetMergedSchemaProperties(GetEntitySchemaPropertiesOptions options) {
+		if (string.IsNullOrWhiteSpace(options.SchemaName)) {
+			throw new EntitySchemaDesignerException("Schema name is required.");
+		}
+		RuntimeEntitySchemaResult runtimeSchema = ReadMergedRuntimeSchema(options.SchemaName.Trim());
+		List<EntitySchemaPropertyColumnInfo> columns = runtimeSchema.Columns
+			.Select(MapRuntimePropertyColumn)
+			.ToList();
+		int inheritedColumnCount = runtimeSchema.Columns.Count(column => column.IsInherited);
```
Comment (tetiana-moshon) [inline clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs:250]: **own/inherited counts have different semantics across the two modes.**

Here merged mode derives `own`/`inherited` from the runtime `IsInherited` flag (i.e. inherited from a parent *entity schema* such as `BaseEntity`). Single-package mode (`GetSchemaProperties`) instead splits on package-layer `Columns` vs `InheritedColumns` — "own" there means *defined in this package layer*. As a result the same column can be classified differently between modes, and `own-column-count`/`inherited-column-count` are not comparable across modes.

This is the same cross-mode-contradiction class the PR deliberately avoids for the nullable schema-level fields, yet the `EntitySchemaPropertiesInfo` remarks state *"All other fields … are authoritative in both modes."* — which slightly overstates it for these two counts and for each column's `source`.

Not a blocker, but please tighten the doc (record remarks + the `GetMergedSchemaProperties` XML doc) to call out that the own/inherited split carries a different meaning in merged mode, so a consumer doesn't compare the counts between a merged and a single-package read.

**Review by b-horodyskyi — COMMENTED** [review]
## Creatio Code Review — PR #678 (re-review of head `a23f958`)

**Decision: APPROVE.** Both Major findings from my prior review (head `535cf47`) are fully resolved by commit `72ece898`, and the two recommended Major test-coverage gaps were added. No new issues. The PR also re-targeted `master` and the ENG-90943 guidance commits dropped from the diff, so the change is now cleanly scoped to the entity-schema feature.

### Resolution of prior findings
- **[RESOLVED] Incomplete exception translation** — `ReadMergedRuntimeSchema` now catches `InvalidOperationException or HttpRequestException or JsonException or TaskCanceledException` via a `when` filter (not a bare catch, per project rule), with a `<remarks>` explaining the MCP-path rationale. New parameterized test `GetSchemaProperties_TranslatesTransportAndParseFailures_WhenPackageIsOmitted` asserts each type is wrapped with the original as `InnerException`. Correct and complete.
- **[RESOLVED] Silent plausible defaults in merged JSON** — chose option (a): `IndexesCount`, `SspAvailable`, `UseRecordDeactivation`, `UseDenyRecordRights`, `UseLiveEditing` are now `int?`/`bool?` and emit `null` in merged mode so a consumer can distinguish "unavailable" from a genuine `false`/`0`. Single-package mode passes real values unchanged (implicit `bool→bool?`/`int→int?` widening — verified compiles). CLI prints `<unknown>` via new `FormatBoolean(bool?)`/`FormatNullableCount(int?)` overloads; tool description, help, and docs updated from "default to null/0/false" to "null = unavailable". Test renamed to `ReportsNullForUnavailableFields_WhenPackageIsOmitted` with null assertions.

### Recommended test gaps — also addressed
- `GetSchemaProperties_LeavesPrimaryColumnNameNull_WhenPrimaryColumnUIdMatchesNoColumn` — covers the orphan `PrimaryColumnUId` branch.
- `GetSchemaProperties_ReturnsEmptyCounts_WhenRuntimeSchemaHasNoColumns` — covers the own/inherited counting boundary at zero columns.

### Remaining (non-blocking)
- `FormatBoolean(bool?)`/`FormatNullableCount(int?)` are duplicated across `GetEntitySchemaPropertiesCommand` and `RemoteEntitySchemaColumnManager` — but this matches the pre-existing duplication of `FormatBoolean(bool)` across both files; not introduced here. Minor.
- Friendly-type characterization test for the known `GetFriendlyTypeName` raw-id fallthrough (Phone/Web) is deferred to the already-spawned follow-up — acceptable, as the behavior is pre-existing and symmetric across both modes.
- `clio.mcp.e2e` merged-mode coverage remains out of CI (documented); CI enforcement rests on the unit tests, which now cover the indexed/title mapping, the nullable-unavailable fields, the orphan-primary and empty-column boundaries, and the transport/parse exception translation.

### Dimension results (unchanged from prior pass, re-confirmed for the delta)
- **Security: PASS** — the change only narrows risk (nullable JSON field + broader fault normalization).
- **Performance: PASS** — single remote round-trip, in-memory projection.
- **Code quality: good** — nullable migration done cleanly, named-arg construction, thorough docs/MCP/help alignment.

### Requirement coverage
- Merged/effective view when `--package` omitted: **covered**.
- Backward-compatible single-package path: **covered** (real non-null values retained).
- Machine-distinguishable unavailable fields: **covered** (null + documentation).
- Normalized failure contract on MCP path: **covered** (broadened translation + tests).
- MCP/docs/help updated: **covered**.

### Confidence limits
Reviewed the fix commit `72ece898` diff plus source at head `a23f958`; the PR branch is not checked out locally and no live Creatio environment was exercised, so acceptance against a real `Account.UsrColumn40` and actual server error-message content remain unverified (noted in the PR's own reviewer notes). Playwright overlay not applicable.

**Review by b-horodyskyi — APPROVED** [review]

File: clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs (hunk 5 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/EntitySchemaDesigner/RemoteEntitySchemaColumnManager.cs:250]: Good catch — fixed in ddedd5c5 (doc-only). Tightened both the EntitySchemaPropertiesInfo remarks and the GetMergedSchemaProperties XML doc to call out that own-column-count/inherited-column-count and each column's source mean different things per mode (runtime parent-entity-schema inheritance in merged mode vs package-layer ownership in single-package mode) and must NOT be compared across modes; corrected the over-broad 'authoritative in both modes' wording.

**Review by tetiana-moshon — APPROVED** [review]
